server/container/guice/guice-common/src/main/java/org/apache/james/utils/GuiceProbeProvider.java

```diff
@@ -0,0 +1,45 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.utils;
+
+import com.github.steveash.guavate.Guavate;
+import com.google.common.base.Preconditions;
+
+import javax.inject.Inject;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public class GuiceProbeProvider {
+    private final Map<Class<GuiceProbe>, GuiceProbe> registry;
+
+    @Inject
+    public GuiceProbeProvider(Set<GuiceProbe> guiceProbes) {
+        this.registry = guiceProbes.stream()
+            .collect(Guavate.toImmutableMap(guiceProbe -> (Class<GuiceProbe>) guiceProbe.getClass()));
+    }
+
+    public <T extends GuiceProbe> T getProbe(Class<T> clazz) {
+        Preconditions.checkNotNull(clazz);
+        return Optional.ofNullable(registry.get(clazz))
+            .map(probe -> (T) probe)
```

Comment:
This seems to work on mine:
public GuiceProbe getProbe(Class<?> clazz) {
        Preconditions.checkNotNull(clazz);
        return Optional.ofNullable(registry.get(clazz))
            .map(GuiceProbe.class::cast)
            .orElseThrow(() -> new RuntimeException("No probe registered for class: " + clazz));
    }
And it silents a warning